pkg/ocm/helpers.go

```diff
@@ -556,7 +556,8 @@ func (c *Client) GetPolicies(policyType string) (map[string]string, error) {
 	return m, nil
 }
 
-func (c *Client) GetCredRequests() (map[string]*cmv1.STSOperator, error) {
+// todo
+func (c *Client) GetCredRequests(isHypershift bool) (map[string]*cmv1.STSOperator, error) {
```

Comment:
I think the parameter should hide the purpose in this case. Maybe just send the cluster object and let this function decide what it needs from the cluster.
```
func (c *Client) GetCredRequests(cluster *cmv1.Cluster) (map[string]*cmv1.STSOperator, error) {
...
```

Reply:
Actually, you're not using `isHypershift` in this function, so maybe just remove it.

Reply:
The implementation is WIP. Still need to figure out how to use this boolean to get the right credential requests. Wanted to make sure that the structure is fine before moving forward to implement this.

Reply:
@vkareh I tried to send the cluster object, but that won't work for the cluster creation; for the cluster creation we need the operatorroles slice, which is created in this block; therefore this piece of code cannot be moved to after the cluster creation.

Reply:
I'm wondering if the cluster spec should have a 'cluster.Provisioner' attribute with 'hive' or 'hypershift' as values, rather than checking 'cluster.HyperShift.Enabled' to determine that it's Hive after all...

Reply:
We had this conversation before, and we voted against that. We feel grouping anything related to hypershift in a struct will simplify adding further information in the future. As an example, we just added a new field (management cluster) in CS.